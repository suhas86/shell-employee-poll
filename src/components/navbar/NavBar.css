.navbar ul {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  text-decoration: none;
}

.navbar li {
  list-style: none;
  padding: 12px;
}

.navbar a {
  text-decoration: none;
  color: #003865;
  font-size: 14px;
}

.navbar a:hover {
  opacity: 0.8;
}

.navbar .active {
  border-bottom: 1px solid #003865;
}
