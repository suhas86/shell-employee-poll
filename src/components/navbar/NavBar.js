import { NavLink } from 'react-router-dom';
import './NavBar.css';

const NavBar = () => {
  return (
    <nav className="navbar">
      <ul>
        <li>
          <NavLink to="/" data-testid="home">
            Home
          </NavLink>
        </li>
        <li>
          <NavLink to="leaderboard" data-testid="leaderboard">
            Leaderboard
          </NavLink>
        </li>
        <li>
          <NavLink to="new" data-testid="new">
            New
          </NavLink>
        </li>
      </ul>
    </nav>
  );
};

export default NavBar;
