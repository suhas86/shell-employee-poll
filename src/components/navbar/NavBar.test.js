import { screen } from '@testing-library/react';
import renderWithRouter from '../../utils/testing/renderWithRouter';
import NavBar from './NavBar';

describe('NavBar', () => {
  it('should render correctly', () => {
    const view = renderWithRouter(<NavBar />);
    expect(view).toMatchSnapshot();
  });
  it('should have all the links', () => {
    renderWithRouter(<NavBar />);
    expect(screen.getByTestId('home')).toBeInTheDocument();
    expect(screen.getByTestId('leaderboard')).toBeInTheDocument();
    expect(screen.getByTestId('new')).toBeInTheDocument();
  });
});
