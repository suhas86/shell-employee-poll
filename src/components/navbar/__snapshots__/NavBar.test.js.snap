// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render correctly 1`] = `undefined`;
